CGE/CGE2: Rewrite keymapper and add joystick support - bugs 15187,15188

bluegr opened the pull request:
This adds keymaps for joystick buttons for CGE, and cleans up the keyboard input code
Fixes bug 15187: https://bugs.scummvm.org/ticket/15187

Edit: Added the same changes for CGE2, as the input code is basically the same as CGE
Fixes bug 15188: https://bugs.scummvm.org/ticket/15188

OMGPizzaGuy reviewed (approved):
Surprised the kEventKeyb only had one use, but this looks fine to me

ccawley2011 commented on engines/cge/metaengine.cpp:
```diff
@@ -200,96 +200,129 @@ Common::KeymapArray CGEMetaEngine::initKeymaps(const char *target) const {
-	act->setKeyEvent(KEYCODE_F1);
+	act->setCustomEngineActionEvent(kActionInfo);
 	act->addDefaultInputMapping("F1");
+	act->addDefaultInputMapping("JOY_LEFT_STICK");
```
```suggestion
	act->addDefaultInputMapping("JOY_LEFT_TRIGGER");
```

ccawley2011 commented on engines/cge/metaengine.cpp:
```diff
@@ -200,96 +200,129 @@ Common::KeymapArray CGEMetaEngine::initKeymaps(const char *target) const {
-	act->setKeyEvent(KeyState(KEYCODE_x, 0, KBD_ALT));
+	act->setCustomEngineActionEvent(kActionQuit);
 	act->addDefaultInputMapping("A+x");
+	act->addDefaultInputMapping("JOY_RIGHT_STICK");
```
```suggestion
	act->addDefaultInputMapping("JOY_RIGHT_TRIGGER");
```

ccawley2011 commented on engines/cge/metaengine.cpp:
```diff
@@ -200,96 +200,129 @@ Common::KeymapArray CGEMetaEngine::initKeymaps(const char *target) const {
+	act->setCustomEngineActionEvent(kActionAltDice);
 	act->addDefaultInputMapping("LALT");
 	act->addDefaultInputMapping("RALT");
+	act->addDefaultInputMapping("JOY_BACK");
```
`JOY_BACK` is already claimed by the global keymap for activating the virtual keyboard.

ccawley2011 commented on engines/cge/metaengine.cpp:
```diff
@@ -200,96 +200,129 @@ Common::KeymapArray CGEMetaEngine::initKeymaps(const char *target) const {
+	keymap->addAction(act);
+
 	// I18N: 3-4 dialogs of game version info, (translation) credits, etc.
 	act = new Action("Game Info", _("Game Info"));
```
I hadn't noticed this before, but the first parameter is supposed to be a short identifier (usually all upper-case without spaces or punctuation) rather than just the untranslated version of the label.

bluegr replied:
You're right. Done

bluegr commented:
@OMGPizzaGuy: Updated with the same changes for CGE2, as its input code is basically the same as CGE
@ccawley2011: Hopefully, I've addressed all of the issues you found

ccawley2011 commented on engines/cge/metaengine.cpp:
```diff
@@ -200,96 +200,129 @@ Common::KeymapArray CGEMetaEngine::initKeymaps(const char *target) const {
+	act->setCustomEngineActionEvent(kActionAltDice);
 	act->addDefaultInputMapping("LALT");
 	act->addDefaultInputMapping("RALT");
+	act->addDefaultInputMapping("JOY_CENTER");
```
`JOY_CENTER` is only available on Android, so it would be better to use something more universal.

```suggestion
	act->addDefaultInputMapping("JOY_Y");
```

bluegr commented:
Thanks for your feedback! Fixed all issues reported. Merging, and closing the two related bugs
